File: java/src/test/java/mx/iteso/ut/QuesoManchegoTest.java
package mx.iteso.ut;

import org.junit.Test;
import org.junit.Before;
import static org.junit.Assert.*;

public class QuesoManchegoTest {
    QuesoManchego quesoManchego;

    @Before
    public void setup(){
        quesoManchego = new QuesoManchego();
    }

    @Test
    public void testTrueMelt(){
        quesoManchego.melt(true);
        assertTrue(quesoManchego.isMelted());
    }

    @Test
    public void testFalseMelt(){
        quesoManchego.melt(false);
        assertFalse(quesoManchego.isMelted());
    }

    @Test
    public void testCurrentTemperature(){
        quesoManchego.setCurrentTemperature(21);
        assertEquals(21, quesoManchego.getCurrentTemperature());
    }

    @Test
    public void testMelting(){
        assertEquals(21, quesoManchego.getMeltingTemperature());
    }
}

File: python/src/unittest/QuesoManchegoTest.py
from QuesoManchego import *
import unittest


class QuesoManchegoTest(unittest.TestCase):
    def setUp(self):
        self.quesoManchego = QuesoManchego()

    def test_CurrentTemperature(self):
        self.quesoManchego.setCurrentTemperature(21)
        self.assertEquals(21, self.quesoManchego.getCurrentTemperature())

    def test_CurrentTemperature(self):
        self.quesoManchego.setCurrentTemperature(21)
        self.assertEquals(21, self.quesoManchego.getCurrentTemperature())

    def test_FalseMelt(self):
        self.quesoManchego.melt(False)
        self.assertFalse(self.quesoManchego.isMelted())

    def test_TrueMelt(self):
        self.quesoManchego.melt(True)
        self.assertTrue(self.quesoManchego.isMelted())

    def test_Melting(self):
        self.assertEquals(20, self.quesoManchego.getMeltingTemperature())
